backend/controllers/notification.controller.js

```diff
@@ -0,0 +1,76 @@
+import { errorHandler } from "../middlewares/error.js";
+import Notification from '../models/notification.model.js';
+;
+
+export const createNotification = async(req,res,next) =>{
+    const notifi = req.body; 
+    try{
+        const notification = new Notification(notifi); 
+        const sample = await notification.save(); 
+        res.status(201).json(sample);
+    }catch (error){
+        next(errorHandler(500, "Failed to create notification"));
+    }
+};  
+
+
+export const getNotifications = async (req, res, next) => {
+    const { acocuntNumber} = req.params;
+    const {page=1, limit= 10} = req.query;
+
+    try {
+        const notifications = await Notification.find({ acocuntNumber})
+            .skip((page - 1) * limit)
+            .limit(parseInt(limit))
+            .sort({ createdAt: -1 }); 
+
+        const totalNotifications = await Notification.countDocuments({ acocuntNumber });
+        const totalPages = Math.ceil(totalNotifications / limit);
+
+        res.send({
+            notifications,
+            totalPages,
+            currentPage: parseInt(page),
+            totalNotifications
+        });
+    } catch (error) {
+        next(errorHandler(500, "Failed to retrieve notifications"));
+    }
+};  
+
+
+export const updateNotification = async(req,res,next)=>{
```

Comment:
is update for single notification? If it is, how are we updating it just by account number

Reply:
why is it that accountNumber is used in findById? shouldn't it be notification id